File: TUnit.TestProject/OptionalArgumentsTests.cs
using TUnit.TestProject.Attributes;

namespace TUnit.TestProject;

[EngineTest(ExpectedResult.Pass)]
public class OptionalArgumentsTests
{
    [Test]
    [Arguments(1)]
    public void Test(int value, bool flag = true)
    {
        // Dummy Method
    }
}

File: TUnit.TestProject.FSharp/OptionalArgumentsTests.fs
namespace TUnit.TestProject.FSharp

open TUnit.Core

// F# equivalent of OptionalArgumentsTests.cs

type OptionalArgumentsTests() =
    [<Test>]
    [<Arguments(1)>]
    member _.Test(value: int, ?flag: bool) =
        let flag = defaultArg flag true
        () // Dummy Method
